### ISSUE TITLE
Unable to resolve type while building object with type

### ISSUE THREAD

**Issue by zh3ngyuan** [ssannandeji/Zenject-2019#290]
I want to bind GameTreePanel with the AdaptiveTest for unit test, and I get no error during compiling, but when I run the test code, there is an error

```
Zenject.ZenjectException : Unable to resolve type 'GameTreePanel[]' while building object with type 'IntegrationTests.AdaptiveTest'. 
Object graph:
AdaptiveTest
```

Here is part of my code:

"AdaptiveTest.cs"

```cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Net;
using Zenject;

using UnityEngine;

namespace IntegrationTests {

[TestFixture]
public class AdaptiveTest : ZenjectIntegrationTestFixture {

    [Inject]
    private ISession session;

    [Inject]
    private DifficultyMap difficultyMap;

    [Inject]
    private GameTreePanel[] gameTreePanels;

    GameRequirements[] gameRequirements = new GameRequirements[6];

    GameModel[] gameModels = new GameModel[6];

    StandardModel[] standardModels = new StandardModel[6];

    GameRoundModel[] gameRoundsModels = new GameRoundModel[6];

    [SetUp]
    public void Init() {

        // Container.Bind<GameTreePanel>().FromNewComponentOnNewGameObject().AsTransient();

        Container.Bind<GameTreePanel>().FromNewComponentOnNewGameObject().AsTransient();
        }
}
```

"GameTreePanel.cs"
```cs

using System.Collections;
using System.Collections.Generic;
using System.Linq;
using UnityEngine;
using UnityEngine.SceneManagement;
using UnityEngine.UI;
using Zenject;

public class GameTreePanel : MonoBehaviour {
	public GameRequirements gameReqs;

	public bool GameLocked {
		get {
			if(difficultyMap != null) {
				return difficultyMap.FindGameReq(gameReqs.gameType).IsLocked;
			}
			return true;
		}
	}

	[SerializeField]
	private GameObject lockedObject;

	[SerializeField]
	private GameObject[] stars;

	private Vector3 mouseClickedPos;
	private float clickThreshold = 10.0f;

	//Zenject
	private FadeUI loadingPanel;
	private UserSubMenu userSubMenu;
	private SurveyManager surveyManager;
	private ISession session;
	private DifficultyMap difficultyMap;
	private LoadingAnimation loadingAnimation;

	[Inject]
	public void Construct(
		FadeUI aLoadingPanel,
		UserSubMenu aUserSubMenu,
		SurveyManager aSurveymanager,
		ISession aSession,
		DifficultyMap aDifficultyMap,
		LoadingAnimation aLoadingAnimation
	) {
		loadingPanel = aLoadingPanel;
		userSubMenu = aUserSubMenu;
		surveyManager = aSurveymanager;
		session = aSession;
		difficultyMap = aDifficultyMap;
		loadingAnimation = aLoadingAnimation;
	}

	#region getters/setters
	public GameObject[] Stars {
		get { return stars; }
	}
	public GameObject LockedObject {
		get { return lockedObject; }
	}
	#endregion

	public void Awake(){
		for (int i = 0; i < stars.Length; i++) {
			Color c = stars[i].GetComponent<SpriteRenderer>().color;
			c = Color.gray;
			stars[i].GetComponent<SpriteRenderer>().color = c;
		}
	}

	void Update(){
	}

	...
}
```

**Comment by svermeulen** [comment]
My best guess is that it's because you're using an array instead of a list to inject GameTreePanel.  Can you try changing it to a `List<GameTreePanel>` instead?

I think we should change Zenject to support arrays though so this could be considered a bug

**Comment by svermeulen** [comment]
Fixed in 6.0
